src/components/spreadsheet/add-new-spreadsheet/add-new-spreadsheet-dialog.tsx

```diff
@@ -0,0 +1,204 @@
+
+import { yupResolver } from '@hookform/resolvers/yup';
+import {
+    getEmptySpreadsheetFormSchema,
+    getSpreadsheetFromModelFormSchema,
+    initialEmptySpreadsheetForm,
+    initialSpreadsheetFromModelForm,
+    SPREADSHEET_MODEL,
+    SPREADSHEET_NAME,
+} from './add-new-spreadsheet-form';
+import { EQUIPMENT_TYPE_FIELD } from 'components/utils/field-constants';
+import { EQUIPMENT_TYPES } from 'components/utils/equipment-types';
+import { NEW_SPREADSHEET_CREATION_OPTIONS } from './constants';
+import { useDispatch, useSelector } from 'react-redux';
+import { updateTableDefinition } from 'redux/actions';
+import { TABLES_DEFINITIONS } from '../utils/config-tables';
+import { AppState } from 'redux/reducer';
+import { getSpreadsheetModel } from 'services/explore';
+import { FormattedMessage } from 'react-intl';
+export type AddSpreadsheetConfigDialogProps = {
+    open: UseStateBooleanReturn;
+    selectedOption: { id: string; label: string } | undefined;
+};
+
+const styles = {
+    dialogContent: {
+        width: '20%',
+        height: '30%',
+        maxWidth: 'none',
+        margin: 'auto',
+    },
+    actionButtons: { display: 'flex', gap: 2, justifyContent: 'end' },
+} as const satisfies Record<string, SxProps<Theme>>;
+
+export default function AddSpreadsheetConfigDialog({
+    open,
+    selectedOption,
+}: Readonly<AddSpreadsheetConfigDialogProps>) {
+    const tablesDefinitionIndexes = useSelector((state: AppState) => state.tables.definitionIndexes);
+    const tablesNames = useSelector((state: AppState) => state.tables.names);
+
+    const emptySpreadsheetFormSchema = useMemo(() => getEmptySpreadsheetFormSchema(tablesNames), [tablesNames]);
+    const spreadsheetFromModelFormSchema = useMemo(() => getSpreadsheetFromModelFormSchema(tablesNames), [tablesNames]);
+
+    const shema: typeof emptySpreadsheetFormSchema | typeof spreadsheetFromModelFormSchema =
+        selectedOption?.id === NEW_SPREADSHEET_CREATION_OPTIONS.EMPTY.id
+            ? emptySpreadsheetFormSchema
+            : spreadsheetFromModelFormSchema;
+
+    const defaultFormValues =
+        selectedOption?.id === NEW_SPREADSHEET_CREATION_OPTIONS.EMPTY.id
+            ? initialEmptySpreadsheetForm
+            : initialSpreadsheetFromModelForm;
+
+    const formMethods = useForm<typeof defaultFormValues>({
+        defaultValues: defaultFormValues,
+        resolver: yupResolver(shema as any),
+    });
+
+    const dispatch = useDispatch();
+    const { handleSubmit, reset } = formMethods;
+
+    const onSubmit = useCallback(
+        (newParams: any) => {
+            if (selectedOption?.id === NEW_SPREADSHEET_CREATION_OPTIONS.EMPTY.id) {
+                const equipmentType = newParams.equipmentType as keyof typeof TABLES_DEFINITIONS;
+                const newTableDefinition = {
+                    ...TABLES_DEFINITIONS[equipmentType],
+                    name: newParams[SPREADSHEET_NAME],
+                    index: tablesDefinitionIndexes.size,
```

Comment:
The created spreadsheet should be added at the beginning of the list of tabs

Reply:
seen with the PO, we can keep it like this for now